test/screenshot/README.md

```diff
@@ -1,22 +1,144 @@
+```
+
+Next, modify a Sass file, and rerun the tests:
+
+```bash
+echo '.mdc-button:not(:disabled){color:red}' >> packages/mdc-button/mdc-button.scss
+npm run screenshot:test
+```
+
+You should see something like this in the terminal:
+
+```
+Found 65 screenshot diffs!
+https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/05/22/16_23_58_618/af3d7271f/report.html
+```
+
+## Basic usage
+
+### Updating "golden" screenshots
+
+```bash
+npm run screenshot:update-goldens
+```
+
+In addition to running the tests, this command will also update your local `test/screenshot/golden.json` file with the 
+newly captured screenshots.
+
+### Rerunning flaky tests
+
+If a single screenshot fails to render, you can rerun it without running the entire suite:
+
+```bash
+npm run screenshot:update-goldens -- \
+  --mdc-include-browser=ie@11 \
+  --mdc-include-url=mdc-button/classes/dense
+```
+
+**IMPORTANT:** Note the `--` between `screenshot:test` and `--help`. This is required by `npm`.
+
+You can rerun multiple screenshots by passing the arguments multiple times:
+
+```bash
+npm run screenshot:update-goldens -- \
+  --mdc-include-browser=ie@11 \
+  --mdc-include-browser=android_chrome \
+  --mdc-include-url=mdc-button/classes/dense \
+  --mdc-include-url=mdc-fab/classes/mini
+```
+
+Partial matches are allowed, so `ie@11` is equivalent to `desktop_windows_ie@11`, and `chrome` is equivalent to 
+`desktop_windows_chrome@latest` and `mobile_android_chrome@latest`.
+
+See `test/screenshot/browser.json` for the full list of supported browsers.
+
+### Local dev server
 
 ```bash
 npm start
 ```
 
 Then open http://localhost:8080/ in your browser.
 
+This replaces the deprecated `npm run dev` command.
+
 Source files are automatically recompiled when they change.
 
+## Advanced usage
+
+To see all available CLI flags, run:
+
+```bash
+npm run screenshot:test -- --help
+```
+
+**IMPORTANT:** Note the `--` between `screenshot:test` and `--help`. This is required by `npm`.
+
+### Diffing against a local `golden.json` file
+
+By default, screenshots are diffed against `origin/master:test/screenshot/golden.json`.
+
+To diff against a local `golden.json` file, run:
+
+```bash
+npm run screenshot:update-goldens -- --mdc-diff-base=test/screenshot/golden.json
+```
+
+URLs are also supported:
+
+```bash
+npm run screenshot:update-goldens -- --mdc-diff-base=https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/05/22/17_34_19_887/c8c29033e/golden.json
+```
+
+### Diffing against another branch
+
+By default, screenshots are diffed against `origin/master:test/screenshot/golden.json`.
+
+To diff against a different branch, run:
+
+```bash
+npm run screenshot:update-goldens -- --mdc-diff-base=fix/fab/icon-alignment-ie11
+```
+### Public demos
+
+```bash
+npm run screenshot:upload-assets
+```
+
+This will upload all test assets (HTML/CSS/JS files) to a public URL and print the URL in the terminal.
```

Comment:
all **currently-built** test assets, I assume... (i.e. you need to run `screenshot:build` first)

Reply:
Now that #2773 is merged, `test` and `update-goldens` automatically run `build` first, so "currently-built" seems a bit redundant; the assets are always built.

Reply:
Yes, `upload-assets` also goes through the `shouldBuild` flow 😀 

All of the `run-*.js` files call `Controller.initialize()`, which automatically builds the assets (unless you pass `--mdc-skip-build` or have `npm start` running in another terminal).